schemas/telemetry/update/update.4.parquetmr.txt

```diff
@@ -1,6 +1,17 @@
 message update {
   required binary id (UTF8);
   required binary clientId (UTF8);
+  required group application {
+    required binary architecture (UTF8);
+    required binary buildId (UTF8);
```

Comment:
I seems to me that a lot of fields are redundant with the one in `environment.build` is that fine?

Reply:
Correct - I wonder if it makes sense to just keep the application.displayVersion at this point. @mreid-moz ?

Reply:
Hmm... it's been a while since I've looked into this, but last time I did, I thought that the values in [`application` ](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/TelemetryController.jsm#355-365) were available more often / more reliably than the ones in [`environment.build`](https://dxr.mozilla.org/mozilla-central/source/toolkit/components/telemetry/TelemetryEnvironment.jsm#1286-1296). If that's so, it's preferable to keep the entire `application` group and remove the dupes from `environment.build`. If not, please let me know so I can update my understanding :)  At the least, the `application` section should be available for all pings whether they include the `environment` or not, so it should be more "standard".

Reply:
Mh, weird: both sections should be there and have identical values for matching fields. Unless there's a bug (in the client), of course :-D There should be no mismatches for pings that report the environment.

With that said, let's just add the full application and remove the dupes from the environment as @mreid-moz said. Mark, I imagine this change would require backfilling, right?

Reply:
Ok, as long as both sets of fields should be the same, and are equally reliable, I am fine with the original approach of just adding `application.displayVersion`. That should avoid the need for backfill and minimize impact on any existing queries / analyses.  Sorry for the back-and-forth on this :-/